graders/c/cgrader/cgrader.py

```diff
@@ -316,21 +325,21 @@ def link_object_files(
 
     def test_compile_file(
         self,
-        c_file,
-        exec_file=None,
-        main_file=None,
-        add_c_file=None,
-        compiler=None,
-        points=1,
-        field=None,
-        flags=None,
-        pkg_config_flags=False,
-        name="Compilation",
-        add_warning_result_msg=True,
-        ungradable_if_failed=True,
-        enable_asan=False,
-        reject_symbols=None,
-        objcopy_args=None,
+        c_file: str | list[str],
+        exec_file: str | None = None,
+        main_file: str | None = None,
+        add_c_file: str | list[str] | None = None,
+        compiler: str | None = None,
+        points: int | float = 1,
```

Comment:
Iirc this is equivalent to just float. Double check but I think float subsumes int for the purpose of type checking.

Reply:
TIL: https://typing.readthedocs.io/en/latest/spec/special-types.html#special-cases-for-float-and-complex.

Reply:
Changed in 56a1ef27fbf245a130ac6d939c1a9187e0c2ca70.